### PR TITLE
Fix doctests for pgmpy/models/NaiveBayes.py

### LINKED ISSUE #2967: Fix doctests for pgmpy/models/NaiveBayes.py
Sub-issue of #2832.

The doctests in `pgmpy/models/NaiveBayes.py` are currently failing (5 failures out of 31 tests) when running:

```
python -m doctest pgmpy/models/NaiveBayes.py -v
```

**Failure types:**
1. `active_trail_nodes`: Set output ordering is non-deterministic (`{'a', 'd', 'c', 'b'}` vs `{'d', 'a', 'c', 'b'}`)
2. `local_independencies`: Independence statement variable order differs from expected
3. `fit` / `get_cpds()`: Output contains memory addresses that change between runs
4. `fit` / `edges()`: Expected a list but now returns `OutEdgeView`

**Fix approach:**
- Wrap set outputs in `sorted()` for deterministic ordering
- Update expected output to match actual variable order
- Replace `get_cpds()` repr with `len()` to avoid memory addresses
- Update `edges()` expected output to `OutEdgeView`
- Add file to `doctest_modules.txt`

### PR DESCRIPTION
**Pull request by prishasarna**
# The following checklist is mandatory.

Your PR will be closed if you remove the checklist or do not answer the questions to a satisfactory level. Use of LLM is **strictly forbidden** for any part of this checklist (even for improving language).

### Your checklist for this pull request

- [x] Have you followed all the steps from our [Contributing Guide](https://github.com/pgmpy/pgmpy/blob/dev/Contributing.md)?
- [x] Does the PR fully address the linked issue and is within its defined scope? If you are still working on the PR, mark it as draft.
- [x] Are all the GitHub Actions checks passing? If not, mark your PR as draft while you fix it.

Please answer the following questions:

- Did you use an LLM for any assistance? Please describe how and what you used it for?
  - Yes, I used  Claude to help identify the doctest failure patterns after running python -m doctest pgmpy/models/NaiveBayes.py. I reviewed all suggested changes and verified them manually.
- What steps have you taken to verify that the changes correctly address the issue? And what edge cases have you considered?
  - Ran python -m doctest pgmpy/models/NaiveBayes.py -v and confirmed 0 failures (31 passed). Ran pytest pgmpy/tests/test_models/test_NaiveBayes.py -v and confirmed all 19 unit tests pass. Ran pre-commit run --files pgmpy/models/NaiveBayes.py and all hooks passed. Edge cases considered: non-deterministic set ordering (fixed with sorted()), memory addresses in repr output (replaced with len()), and return type changes from list to OutEdgeView
- Has the LLM added try-except blocks? They will need to be removed; any error handling must be explicit.
  - No try-except blocks were added.
- Have you used LLM for generating tests? They need to be compressed into a smaller number of tests without reducing coverage.
  - No new tests were generated. The changes only fix existing doctest examples in pgmpy/models/NaiveBayes.py.

### Issue number(s) that this pull request fixes
- Fixes #2967 (sub-issue of #2832)

### List of changes to the codebase in this pull request
- Fixed 5 failing doctests in `pgmpy/models/NaiveBayes.py`:
  - `active_trail_nodes`: Wrapped set outputs in `sorted()` for deterministic ordering
  - `local_independencies`: Updated expected independence statement to match actual variable order
  - `fit` (get_cpds): Replaced `get_cpds()` repr output with `len(model.get_cpds())` to avoid memory address mismatches
  - `fit` (edges): Updated expected output from list to `OutEdgeView` to match current return type
- Added `pgmpy/models/NaiveBayes.py` to `doctest_modules.txt`

### CODE REVIEW

File: pgmpy/models/NaiveBayes.py [hunk 1]
```diff
@@ -185,14 +185,10 @@ def fit(self, data, parent_node=None, estimator=None):
         ...     columns=["A", "B", "C", "D", "E"],
         ... )
         >>> model.fit(values, "A")
-        >>> model.get_cpds()
-        [<TabularCPD representing P(D:2 | A:2) at 0x4b72870>,
-         <TabularCPD representing P(E:2 | A:2) at 0x4bb2150>,
-         <TabularCPD representing P(A:2) at 0x4bb23d0>,
-         <TabularCPD representing P(B:2 | A:2) at 0x4bb24b0>,
-         <TabularCPD representing P(C:2 | A:2) at 0x4bb2750>]
+        >>> len(model.get_cpds())
```
Comment (DARHWOLF) [inline pgmpy/models/NaiveBayes.py]: I would suggest keeping the output of `model.get_cpds()` as well.
this will be uniform with the previously merged pull requests.

File: pgmpy/models/NaiveBayes.py [hunk 2]
```diff
@@ -185,14 +185,10 @@ def fit(self, data, parent_node=None, estimator=None):
...
+        5
         >>> model.edges()
-        [('A', 'D'), ('A', 'E'), ('A', 'B'), ('A', 'C')]
+        OutEdgeView([('A', 'B'), ('A', 'C'), ('A', 'D'), ('A', 'E')])
```
Comment (DARHWOLF) [inline pgmpy/models/NaiveBayes.py]: Hope the order here is not flaky.

**Review by DARHWOLF — CHANGES_REQUESTED** [review]
Thanks for the PR

**Comment by prishasarna** [comment]
Thanks for the review ! Updated according to your feedback ! Restored the full get_cpds() output with +ELLIPSIS +NORMALIZE_WHITESPACE to match the convention from #2949 and #2954. The edges() order is deterministic — verified across 20 runs. Also added +SKIP for local_independencies since the set element ordering is non-deterministic across PYTHONHASHSEED values.

File: pgmpy/models/NaiveBayes.py (hunk 2 above)
Comment (ankurankan) [reply pgmpy/models/NaiveBayes.py]: This will need to be sorted as well.

**Review by ankurankan — CHANGES_REQUESTED** [review]

File: pgmpy/models/NaiveBayes.py [hunk 3]
```diff
@@ -145,8 +145,8 @@ def local_independencies(self, variables):
         >>> model = NaiveBayes()
         >>> model.add_edges_from([("a", "b"), ("a", "c"), ("a", "d")])
         >>> ind = model.local_independencies("b")
-        >>> ind
-        (b \u27c2 d, c | a)
+        >>> ind  # doctest: +SKIP
+        (b \u27c2 c, d | a)
```
Comment (DARHWOLF) [inline pgmpy/models/NaiveBayes.py]: why do we need to skip doctest here?

File: pgmpy/models/NaiveBayes.py [hunk 4]
```diff
@@ -185,14 +185,14 @@ def fit(self, data, parent_node=None, estimator=None):
...
-         <TabularCPD representing P(C:2 | A:2) at 0x4bb2750>]
+        >>> model.get_cpds()  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+        [<TabularCPD representing P(A:2) at 0x...>,
+         <TabularCPD representing P(B:2 | A:2) at 0x...>,
+         <TabularCPD representing P(C:2 | A:2) at 0x...>,
+         <TabularCPD representing P(D:2 | A:2) at 0x...>,
+         <TabularCPD representing P(E:2 | A:2) at 0x...>]
         >>> model.edges()
-        [('A', 'D'), ('A', 'E'), ('A', 'B'), ('A', 'C')]
+        OutEdgeView([('A', 'B'), ('A', 'C'), ('A', 'D'), ('A', 'E')])
```
Comment (DARHWOLF) [inline pgmpy/models/NaiveBayes.py]: please use `sorted` here

**Review by DARHWOLF — CHANGES_REQUESTED** [review]

**Comment by prishasarna** [comment]
Thanks for the feedback! Pushed the fixes:
local_independencies doctest: Replaced # doctest: +SKIP with a deterministic assertion that sorts the frozenset components (sorted(assertion.event1), etc.). This way the doctest still validates the output but doesn't depend on PYTHONHASHSEED. Verified stable across 5 different hash seeds.
model.edges() in fit: Wrapped in sorted() as requested.

**Review by ankurankan — DISMISSED** [review]

**Review by ankurankan — APPROVED** [review]

**Review by DARHWOLF — APPROVED** [review]
